crates/fluvio/src/producer/partition_producer.rs

```diff
@@ -0,0 +1,238 @@
+pub(crate) struct PartitionProducer {
+    replica: ReplicaKey,
+    spu_pool: Arc<SpuPool>,
+    batches_lock: Arc<Mutex<VecDeque<ProducerBatch>>>,
+    batch_events: Arc<BatchEvents>,
+    linger: Duration,
+    last_error: Arc<RwLock<Option<ProducerError>>>,
+}
+
+impl PartitionProducer {
+    fn new(
+        replica: ReplicaKey,
+        spu_pool: Arc<SpuPool>,
+        batches_lock: Arc<Mutex<VecDeque<ProducerBatch>>>,
+        batch_events: Arc<BatchEvents>,
+        linger: Duration,
+        last_error: Arc<RwLock<Option<ProducerError>>>,
+    ) -> Self {
+        Self {
+            replica,
+            spu_pool,
+            batches_lock,
+            batch_events,
+            linger,
+            last_error,
+        }
+    }
+
+    pub fn shared(
+        replica: ReplicaKey,
+        spu_pool: Arc<SpuPool>,
+        batches: Arc<Mutex<VecDeque<ProducerBatch>>>,
+        batch_events: Arc<BatchEvents>,
+        linger: Duration,
+        error: Arc<RwLock<Option<ProducerError>>>,
+    ) -> Arc<Self> {
+        Arc::new(PartitionProducer::new(
+            replica,
+            spu_pool,
+            batches,
+            batch_events,
+            linger,
+            error,
+        ))
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub(crate) fn start(
+        replica: ReplicaKey,
+        spu_pool: Arc<SpuPool>,
+        batches: Arc<Mutex<VecDeque<ProducerBatch>>>,
+        batch_events: Arc<BatchEvents>,
+        linger: Duration,
+        error: Arc<RwLock<Option<ProducerError>>>,
+        end_event: Arc<StickyEvent>,
+        flush_event: (Arc<EventHandler>, Arc<EventHandler>),
+    ) {
+        let producer =
+            PartitionProducer::shared(replica, spu_pool, batches, batch_events, linger, error);
+        fluvio_future::task::spawn(async move {
+            producer.run(end_event, flush_event).await;
+        });
+    }
+
+    #[instrument(skip(self, end_event, flush_event))]
+    async fn run(
+        &self,
+        end_event: Arc<StickyEvent>,
+        flush_event: (Arc<EventHandler>, Arc<EventHandler>),
+    ) {
+        use tokio::select;
+
+        let mut linger_sleep = sleep(std::time::Duration::from_secs(1800));
+
+        loop {
+            select! {
+                _ = end_event.listen() => {
+                    info!("partition producer end event received");
+                    break;
+                },
+                _ = flush_event.0.listen() => {
+
+
+                    debug!("flush event received");
+                    if let Err(e) = self.flush(true).await {
+                        error!("Failed to flush producer: {}", e);
+                        self.set_error(e).await;
+                        flush_event.1.notify().await;
+
+                        continue;
+                    }
+                    flush_event.1.notify().await;
+                    linger_sleep = sleep(std::time::Duration::from_secs(1800));
```

Comment:
get value from config

Reply:
every time that there is a manual flush, there is not need to wait for the linger time because all is supposed to be removed from the accumulator. This is just setting the linger time as something large. This is only changed to the config value once we get a new batch event